src/pymor/parameters/base.py

```diff
@@ -572,6 +572,59 @@ def sample_randomly(self, count=None):
         else:
             return [get_param() for _ in range(count)]
 
+    def sample_logarithmic_uniformly(self, counts):
+        """Logarithmically uniform sample |parameter values| from the space.
+
+        Parameters
+        ----------
+        counts
+            Number of samples to take per parameter and component
+            of the parameter. Either a dict of counts per |Parameter|
+            or a single count that is taken for each parameter in |Parameters|.
+
+        Returns
+        -------
+        List of |parameter value| dicts.
+        """
+        if not isinstance(counts, dict):
+            counts = {k: counts for k in self.parameters}
+
+        logspaces = tuple(np.geomspace(self.ranges[k][0],
+                                       self.ranges[k][1],
+                                       num=counts[k])
+                          for k in self.parameters)
+        iters = tuple(product(ls, repeat=max(0, np.zeros(sps).size))
+                      for ls, sps in zip(logspaces, self.parameters.values()))
+        return [Mu((k, np.array(v)) for k, v in zip(self.parameters, i))
+                for i in product(*iters)]
+
+    def sample_logarithmic_randomly(self, count=None, pool_multiplier=100):
+        """Logarithmically scaled random sample |parameter values| from the space.
+
+        Parameters
+        ----------
+        count
+            If `None`, a single dict `mu` of |parameter values| is returned.
+            Otherwise, the number of logarithmically random samples to generate and return as
+            a list of |parameter values| dicts.
+        pool_multiplier
+            Determines how big of a pool the samples are selected.
+
+        Returns
+        -------
+        The sampled |parameter values|.
+        """
+        def get_param():
+            return Mu(((k, np.random.choice(np.geomspace(self.ranges[k][0],
+                                                         self.ranges[k][1],
+                                                         pool_multiplier*size),
```

Comment:
Sorry @Congpy for answering so late.

I do not really like that the sampling is dependent on a discretization of the sampling range. @pmli, correct me if I am wrong, but I think the right approach would be to do something like

```python
np.exp(get_rng().uniform(np.log(self.ranges[k][0]), np.log(self.ranges[k][1]), size))
```
This should give you a distribution on the parameter range with density $\frac{1}{\ln(r) - \ln(l)} \cdot \frac{1}{x}$, and this density should be independent of the choice of base of the logarithm.

Reply:
Looks good to me. I also think it shouldn't depend on discretization.

TIL that this is called a [reciprocal (or log-uniform) distribution](https://en.wikipedia.org/wiki/Reciprocal_distribution). It is available in [`scipy.stats`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.stats.loguniform.html), but not in NumPy.